control/autopilot/include/autopilot/autopilot_inl.h

```diff
@@ -1147,20 +1222,57 @@ quadrotor_common::ControlCommand AutoPilot::executeTrajectory(
       *trajectory_execution_left_duration = ros::Duration(0.0);
       *trajectories_left_in_queue = 0;
       trajectory_queue_.pop_front();
-      setAutoPilotState(States::HOVER);
-      return base_controller_.run(state_estimate, reference_state_,
+      setAutoPilotStateForced(States::HOVER);
```

Comment:
This should not be forced to avoid jumpy behavior if for whatever reason the trajectory ends with a non zero velocity. So by not forcing it will do a smooth breaking if necessary

Reply:
Yes but the you do not guarantee that you reach the end state...
the end of a trajectory normally has zero velocity ;-) and even if not it just means that you get to a stop with an offset where the velocity error is compensated by the position error.

I propose to change `setAutoPilotStateForced()` so that it sets reference state velocity to zero when trigger to hover.